src/zenml/orchestrators/step_launcher.py

```diff
@@ -271,107 +273,78 @@ def launch(self) -> StepRunResponse:
+            pipeline_run=pipeline_run,
+            stack=self._stack,
+        )
+        dynamic_config = self._step if self._snapshot.is_dynamic else None
+        step_run_request = request_factory.create_request(
+            invocation_id=self._invocation_id,
+            dynamic_config=dynamic_config,
+        )
+        step_run_request.logs = logs_request
 
-            try:
-                request_factory.populate_request(request=step_run_request)
-            except BaseException as e:
-                logger.exception(
-                    f"Failed preparing step `{self._invocation_id}`."
-                )
-                step_run_request.status = ExecutionStatus.FAILED
-                step_run_request.end_time = utc_now()
-                step_run_request.exception_info = (
-                    exception_utils.collect_exception_information(e)
+        try:
+            request_factory.populate_request(request=step_run_request)
+        except BaseException as e:
+            logger.exception(f"Failed preparing step `{self._invocation_id}`.")
+            step_run_request.status = ExecutionStatus.FAILED
+            step_run_request.end_time = utc_now()
+            step_run_request.exception_info = (
+                exception_utils.collect_exception_information(e)
+            )
+            raise
+        finally:
+            step_run = Client().zen_store.create_run_step(step_run_request)
+            self._step_run = step_run
+            if model_version := step_run.model_version:
+                step_run_utils.log_model_version_dashboard_url(
+                    model_version=model_version
                 )
-                raise
-            finally:
-                step_run = Client().zen_store.create_run_step(step_run_request)
-                self._step_run = step_run
-                if model_version := step_run.model_version:
-                    step_run_utils.log_model_version_dashboard_url(
-                        model_version=model_version
-                    )
 
             if not step_run.status.is_finished:
                 logger.info(f"Step `{self._invocation_id}` has started.")
 
-                try:
-                    # here pass a forced save_to_file callable to be
-                    # used as a dump function to use before starting
-                    # the external jobs in step operators
-                    if isinstance(
-                        logs_context,
-                        step_logging.PipelineLogsStorageContext,
-                    ):
-                        force_write_logs = (
-                            logs_context.storage.send_merge_event
+                logs_context = nullcontext()
+                if step_run.logs:
+                    logs_context = LoggingContext(log_model=step_run.logs)
+
+                with logs_context:
+                    try:
+                        # TODO: We still need to apply the fix for step operators here
```

Comment:
This is an open question as well. Currently, this is not a problem as we do not have a merge mechanism for the artifact log store. But we need to think about this.